.github/workflows/ci.yml

```diff
@@ -1,6 +1,12 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+      - 1.*
+      - 2.*
+  pull_request:
```

Comment:
In this valid yaml. Does it need one list element like `- *`?

Reply:
Plan to keep it as is, 'Valid YAML!' according to https://www.yamllint.com/

Looks like an alternative expression would be to indicate `null` explicitly.
```yaml
on:
  push:
    branches:
      - main
      - 1.*
      - 2.*
  pull_request: null
env:
```
^ At least that is what a linter did.